manager/integration/tests/test_basic.py

```diff
@@ -4418,7 +4419,96 @@ def test_space_usage_for_rebuilding_only_volume():  # NOQA
+    Test case: the normal scenario
+    1. Create a new volume with 3Gi spec size.
+    2. Write 3Gi data (using `dd`) to the volume.
+    3. Take a snapshot then mark this snapshot as Removed.
+       (this snapshot won't be deleted immediately.)
+    4. Write 3Gi data (using `dd`) to the volume again.
+    5. Delete a random replica to trigger the rebuilding.
+    6. Wait for the rebuilding complete. And verify the volume actual size
+       won't be greater than 2x of the volume spec size.
+    7. Delete the volume.
+
+    """
+    # Pre condition
+    disk_volname = "disk-vol"
+    lht_hostId = get_self_host_id()
+    node = client.by_id_node(lht_hostId)
+    extra_disk_path = create_host_disk(client, disk_volname,
+                                       str(7 * Gi), lht_hostId)
+
+    extra_disk = {"path": extra_disk_path, "allowScheduling": True}
+    update_disks = get_update_disks(node.disks)
+    update_disks["extra-disk"] = extra_disk
+    node = update_node_disks(client, node.name, disks=update_disks,
+                             retry=True)
+    node = common.wait_for_disk_update(client, lht_hostId,
+                                       len(update_disks))
+
+    for fsid, disk in iter(node.disks.items()):
+        if disk.path != extra_disk_path:
+            disk.allowScheduling = False
+            update_disks = get_update_disks(node.disks)
+            update_node_disks(client, node.name, disks=update_disks,
+                              retry=True)
+            node = wait_for_disk_status(client, node.name,
+                                        fsid,
+                                        "allowScheduling", False)
+
+            break
+
+    # The normal scenario
+    volume = create_and_check_volume(client, volume_name, size=str(3 * Gi))
+    volume.attach(hostId=lht_hostId)
+    volume = common.wait_for_volume_healthy(client, volume_name)
+
+    snap_offset = 1
+    volume_endpoint = get_volume_endpoint(volume)
+    write_volume_dev_random_mb_data(volume_endpoint,
+                                    snap_offset, 3000)
+
+    snap2 = create_snapshot(client, volume_name)
+    volume.snapshotDelete(name=snap2.name)
+    volume.snapshotPurge()
+    wait_for_snapshot_purge(client, volume_name, snap2.name)
+
+    write_volume_dev_random_mb_data(volume_endpoint,
+                                    snap_offset, 3000)
+
+    for r in volume.replicas:
+        if r.hostId != lht_hostId:
+            volume.replicaRemove(name=r.name)
+            break
+
+    wait_for_volume_degraded(client, volume_name)
+    wait_for_rebuild_start(client, volume_name)
+    wait_for_rebuild_complete(client, volume_name)
+    volume = client.by_id_volume(volume_name)
+
+    actual_size = int(volume.controllers[0].actualSize)
+    spec_size = int(volume.size)
+
+    assert actual_size/spec_size <= 2
+
+    def finalizer():
+        client.delete(volume)
+        wait_for_volume_delete(client, volume.name)
+
+    request.addfinalizer(finalizer)
```

Comment:
Do we need this part?

Reply:
Yes need clean this volume first then `cleanup_test_disks` in `cleanup_client` can work properly

Reply:
Is it possible to rearrange the sequence in `cleanup_client`?

Reply:
Hi @c3y1huang , 

In this test case there exist 2 volumes, 1 is `vol-disk` created in prepare stage and used as disk, another one is test volume which attached to `vol-disk` when test case execute. After test complete, the teardown function `cleanup_all_volumes`will delete test volume first(because volume name started with letter "L", then delete disk volume `vol-disk`), but function just execute `client.delete(v)` sequentially in a very short time. This will cause test volume stuck at deleting state forever because we should wait test volume is really gone then delete the disk volume, that's why the `wait_for_volume_delete(client, volume.name)` I added in the end of test case

I did local test to rename volume name let test volume be deleted first also reorder `cleanup_test_disks` `cleanup_all_volumes` but not really help, or should I refactor `cleanup_all_volumes`?  thank you.

Reply:
Will it make a difference adding `wait_for_volume_delete` in `cleanup_all_volumes` after `client.delete(v)`?

Reply:
HI @c3y1huang 
I updated PR code and have e2e test reult below, looks like new code does not have soide effect thank you
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2554/
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2553/

Reply:
Suggest splitting to another commit for the cleanup fix.

Reply:
Hi @c3y1huang , please see updated code and https://github.com/longhorn/longhorn-tests/pull/1193 thank you.

Reply:
Sorry I misunderstood the suggestion, will follow the instruction, thank you